File: 1695 - Maximum Erasure Value/solution.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& A) {
        unordered_set<int> s;
        int res = INT_MIN, total = 0, l = 0;
        for(int i = 0; i < A.size(); i++){
            while(s.count(A[i])){
                s.erase(A[l]);
                total -= A[l++];
            }
            s.insert(A[i]);
            total += A[i];
            res = max(res, total);
        }
        return res;
    }
};

File: 1695 - Maximum Erasure Value/solution.py
class Solution:
    def maximumUniqueSubarray(self, A: List[int]) -> int:
        res, total, l, seen = -inf, 0, 0, set()
        for i, v in enumerate(A):
            while v in seen:
                seen.remove(A[l])
                total -= A[l]; l += 1
            seen.add(v)
            total += v
            res = max(res, total)
        return res